astropy/modeling/tabular.py

```diff
@@ -180,13 +210,23 @@ def evaluate(self, *inputs):
             Input coordinates. The number of inputs must be equal
             to the dimensions of the lookup table.
         """
+        if isinstance(inputs, u.Quantity):
+            inputs = inputs.value
+
         inputs = [inp.flatten() for inp in inputs[: self.n_inputs]]
         inputs = np.array(inputs).T
-        if not has_scipy:
+        if not has_scipy:  # pragma: no cover
             raise ImportError("This model requires scipy >= v0.14")
-        return interpn(self.points, self.lookup_table, inputs,
-                       method=self.method, bounds_error=self.bounds_error,
-                       fill_value=self.fill_value)
+        result = interpn(self.points, self.lookup_table, inputs,
+                         method=self.method, bounds_error=self.bounds_error,
+                         fill_value=self.fill_value)
+
+        # return_units not respected when points has no units
```

Comment:
Note: I thought `return_units` would kick in and it doesn't matter that `points` has no units, but that is not the case currently. What is going on here? I am using it wrong?

Reply:
My understanding is that this is how it is supposed to work. Having return_units only makes sense if you know the units of the input. @astrofrog can confirm this.

Reply:
In the case of 2D lookup table, there is a use case (though not in synphot) where you can provide indices (pixel locations) as points and flux as lookup table. The `lookup_table` already has unit, however it is lost if inputs don't. In the case of tabular models, output unit does not rely on input unit. So if `return_units` is not the right thing to use, then what is?

Here is what happens when I remove this logic:
```python
>>> import numpy as np
>>> import astropy.units as u
>>> from astropy.modeling import models
>>> m = models.Tabular2D(lookup_table=np.random.random((10,10)) * u.nJy)
>>> print(m)
Model: Tabular2D
Inputs: ('x0', 'x1')
Outputs: ('y',)
Parameters: 
  points: (array([ 0.,  1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  9.]), array([ 0.,  1.,  2.,  3.,  4.,  5.,  6.,  7.,  8.,  9.]))
  lookup_table: [[ 0.64457699  0.78894069  0.1156456   0.37876596  0.06993314  0.96655462
   0.26624813  0.60061217  0.86790448  0.72952123]
 [ 0.74107928  0.14495951  0.151671    0.48996219  0.63837891  0.738999
   0.0018959   0.38887397  0.85494006  0.05586194]
 [ 0.72020257  0.40558118  0.90156257  0.73898652  0.84281505  0.98399963
   0.6978882   0.79763857  0.64438813  0.98370145]
 [ 0.06835533  0.99982625  0.17347585  0.29238522  0.05112954  0.18628471
   0.45593466  0.22066081  0.33717252  0.43847504]
 [ 0.86884535  0.22043719  0.96749657  0.8592954   0.63047391  0.80161368
   0.59674124  0.87366194  0.72421939  0.0737394 ]
 [ 0.57545182  0.63113073  0.79681753  0.31114323  0.74779429  0.92353172
   0.49895507  0.89662567  0.83687091  0.34031391]
 [ 0.78364421  0.11940549  0.24802695  0.05590922  0.41563473  0.41170051
   0.44405022  0.83144576  0.92452118  0.27208665]
 [ 0.97020015  0.03232788  0.35192894  0.27759741  0.57590458  0.94867373
   0.70884531  0.68262355  0.5200224   0.55835306]
 [ 0.31192752  0.58378825  0.24470681  0.61993295  0.4085091   0.70748766
   0.32441952  0.06683886  0.11489755  0.11240237]
 [ 0.78956128  0.6259519   0.15049759  0.06385817  0.79928124  0.82439344
   0.17882733  0.83831423  0.06880644  0.98039448]] nJy
  method: linear
  fill_value: nan
  bounds_error: True

>>> m([0.5, 0.5], [0, 1])
 array([ 0.69282814,  0.4669501 ])
```

Note that I can force the result to have unit again by providing *some kind* of unit to inputs. However, in this case, assigning `pix` to indices looks kinda misleading:
```python
>>> m([0.5, 0.5] * u.pix, [0, 1] * u.pix)
<Quantity [ 0.69282814, 0.4669501 ] nJy>
```

Reply:
I see. I have no strong feelings either way, that is the explanation why `return_units` did not work. There may be reasonable exceptions. If I'm not mistaken there are other models which force `return_units`. I'm fine with putting that logic back. If you do so please add a note in the docstrings about this.